server/main_server/src/main.cpp

```diff
@@ -734,6 +737,17 @@ Mandatory arguments to long options are mandatory for short options too.
             return -1;
         }
 
+        // SIGUSR2 is used by the agent factory to notify the main server process
+        // when it has completed initialization.
+        struct sigaction sa_sigusr2; // NOLINT(cppcoreguidelines-pro-type-member-init)
+        sigemptyset(&sa_sigusr2.sa_mask);
+        sa_sigusr2.sa_flags = 0;
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-union-access)
+        sa_sigusr2.sa_handler = [](int) { g_agent_factory_initialized = 1; }; // TODO Should we constrain this to the agent factory?
+        if (sigaction(SIGUSR2, &sa_sigusr2, nullptr) == -1) {
+            return -1;
+        }
```

Comment:
Handle TODO comment.

We can adjust the signal handler to only react to signals from the agent factory. We do that kind of thing in the agent factory code because it makes sense for it to only accept signals from its parent.

It isn't as obvious whether the main server process should only react to this signal from its child.

Reply:
if the main server can react to sigusr2 from any of its family... we might be able to use that later from a control plane or GUI or something.  but if we don't have a clear use case today, and we can't imagine it being a blocker or headache to limit it to the agent factory, then it seems helpful to constrain it today?

Reply:
> Is it when the agent factory is able to take a client request? Or maybe it's when the agent factory process has a PID in the process tree?

I think we're most interested in ability to accept requests... the process tree thing would presumably happen 'before' being able to accept requests.. and so therefore would be a 'wider' window... which is less what I think we want than 'ready'.

Reply:
that does make sense.   not a big threat to leave it open.

Reply:
Ok. I'm going to leave it open until someone raises an issue.